carbonserver/trie.go

```diff
@@ -2221,17 +2230,24 @@ func (ti *trieIndex) getNodeFullPath(node *trieNode) string { // skipcq: SCC-U10
 
 	return ""
 }
-
-// skipcq: RVV-A0005
-func (ti *trieIndex) throttle(ps *points.Points, inCache bool) bool {
-	dataLen := int64(len(ps.Data))
-	if dataLen == 0 {
-		// WHY: in theory, this should not happen. But in cases where
-		// go-carbon receives a metric without data points, it should
-		// still be counted as one data point, for throttling and
-		// throughput accounting.
-		dataLen = 1
+func (ti *trieIndex) maxCreatesThrottle(ps *points.Points) bool {
+	select {
+	case keep, open := <-ti.maxCreatesTicker.C:
+		if keep || !open {
+			return false
+		}
+		break
+	default:
+		break
 	}
+	atomic.AddUint64(&ti.throttledCreates, 1)
+	ti.logger.Error("metric creation throttled",
```

Comment:
the amount of generated logs will correlate with the incoming traffic. May I suggest replacing this by a Prometheus counter and removing the logs?